### PR TITLE
fix php 7.2 warnings #121

### PR DESCRIPTION
**Pull request by NF13**

### CODE REVIEW

File: contrib.php [hunk 1]
```diff
@@ -15,6 +15,12 @@
 // multiuser permission check
 permission_or_die(PERM_WRITE);
 
+function arrayEach(&$arr) {
```
Comment (andig) [inline contrib.php]: while this may work, I wouldn't trust a function that I can't grasp by looking at it. This seems like a maintenance nightmare.

File: contrib.php [hunk 2]
```diff
@@ -39,7 +45,7 @@ function multidimsort($array_in, $column)
     reset($multiarray);
 
     // traverse new array of index values and add the corresponding element of the input array to the correct position in the output array
-    while (list ($key, $val) = each ($multiarray))
+    while (list ($key, $val) = arrayEach ($multiarray))
```
Comment (andig) [inline contrib.php]: why not simply use a foreach loop here?

File: core/functions.core.php [hunk 3]
```diff
@@ -277,7 +277,11 @@ function redirect($dest)
  */
 function array_extract($ary, $columnKey)
 {
-    return array_map(create_function('&$row', 'return $row["'.$columnKey.'"];'), $ary);
```
Comment (andig) [inline core/functions.core.php]: did you check if array_column wouldn't do the job?

File: core/template.php [hunk 4]
```diff
@@ -235,6 +235,11 @@ function tpl_filters($filter, $showtv)
     $smarty->assign('listcolumns', session_get('listcolumns'));
 }
 
+function adultcheck_for_video($video)
```
Comment (andig) [inline core/template.php]: unrelated?

File: core/template.php [hunk 5]
```diff
@@ -244,67 +249,76 @@ function tpl_list($list)
 {
     global $smarty, $config;
     global $listcolumns;
-
```
Comment (andig) [inline core/template.php]: a lot of unrelated changes?

Comment (NF13) [reply core/template.php]: count didnt work on null or undefined list in php 7.2 you get a warning see https://www.php.net/manual/function.count.php so you got an warning on the search-start site

File: core/template.php (hunk 4 above)
Comment (NF13) [reply core/template.php]: see line 306 needed for replace of create_function

File: core/template.php (hunk 5 above)
Comment (andig) [reply core/template.php]: I feel it would be cleaner to:

    $smarty->assign('totalresults', 0);

and then return

    if (!$list) { return }

Then you avoid the additional level of indents and checking

Comment (NF13) [reply core/template.php]: changed

File: contrib.php (hunk 1 above)
Comment (andig) [reply contrib.php]: drop this function now?

File: core/functions.core.php (hunk 3 above)
Comment (andig) [reply core/functions.core.php]: this can be dropped now?

File: contrib.php (hunk 2 above)
Comment (andig) [reply contrib.php]: last remaining comment before merging- thanks for your patience ;)

File: core/functions.core.php (hunk 3 above)
Comment (NF13) [reply core/functions.core.php]: yes its removed now

File: core/template.php [hunk 6]
```diff
@@ -244,7 +249,11 @@ function tpl_list($list)
...
+    $smarty->assign('totalresults', 0);
```
Comment (andig) [inline core/template.php]: Move inside the next block as minor optimization if you want

File: contrib.php (hunk 1 above)
Comment (NF13) [reply contrib.php]: removed

File: contrib.php (hunk 2 above)
Comment (NF13) [reply contrib.php]: fixed

File: core/template.php (hunk 6 above)
Comment (NF13) [reply core/template.php]: ✔️

File: contrib.php [hunk 7]
```diff
@@ -39,10 +38,11 @@ function multidimsort($array_in, $column)
     reset($multiarray);
 
     // traverse new array of index values and add the corresponding element of the input array to the correct position in the output array
-    while (list ($key, $val) = each ($multiarray))
+    while (current($multiarray))
```
Comment (andig) [inline contrib.php]: Again very complex- why not use a simple foreach loop instead?

Comment (NF13) [reply contrib.php]: sry my failure you need the key of the array-Element and i has no idea how to get it is this better?

Comment (andig) [reply contrib.php]: You can do this:

    Foreach($foo as $key => $val) 

and ignore $val if you dont need it. Sorry, I could have been clearer here.

File: contrib.php [hunk 8]
```diff
@@ -38,11 +38,10 @@ function multidimsort($array_in, $column)
     reset($multiarray);
```
Comment (andig) [inline contrib.php]: No longer needed with foreach

File: contrib.php (hunk 7 above)
Comment (NF13) [reply contrib.php]: thanks changed

File: contrib.php (hunk 8 above)
Comment (andig) [reply contrib.php]: ... and the code above looks like another use for array_column. I think you can simplify the entire thing and also get rid of the loopvalue (just append to the array instead).

Comment (NF13) [reply contrib.php]: yes you're right

**Comment by NF13** [comment]
@andig the open change request now all fixed if you have any other improvment you are welcome.

File: contrib.php [hunk 9]
```diff
@@ -15,31 +15,20 @@
...
     asort($multiarray);
 
-    // reset internal pointer to beginning of array after above sort
-    reset($multiarray);
-
     // traverse new array of index values and add the corresponding element of the input array to the correct position in the output array
-    while (list ($key, $val) = each ($multiarray))
+    foreach ($multiarray as $key => $val)
     {
         $array_out[$loopvalue] = $array_in[$key];
```
Comment (andig) [inline contrib.php]: One last thing... pls remove the loopvalue var and just do

    $Ary[] = $appendedvalue

And I do realize I wrote much of this code ...

Comment (NF13) [reply contrib.php]: @andig Jo ist angepasst, das hast du ganz schön verkackt 😉 Kleiner Scherz! 😉

So viel sollte ja jetzt nicht mehr von der Methode übrig sein was man optimieren kann 😉

File: contrib.php [hunk 10]
```diff
@@ -15,34 +15,21 @@
...
-    // reset internal pointer to beginning of array after above sort
-    reset($multiarray);
-
     // traverse new array of index values and add the corresponding element of the input array to the correct position in the output array
-    while (list ($key, $val) = each ($multiarray))
+    foreach ($multiarray as $key => $val)
     {
-        $array_out[$loopvalue] = $array_in[$key];
-        $loopvalue++;
+        $array_out[] = $array_in[$key];
```
Comment (andig) [inline contrib.php]: Sagmal- ich sehs erst jetzt- aber ist das hier nicht noch ein array_column?!

Und zu meiner Entschuldigung: die Funktion gabs früher nicht...

Comment (andig) [reply contrib.php]: Mhhm, oder auch nicht. Wir lassen das so 👍🏻

**Comment by andig** [comment]
Super, vielen Dank für Deine Geduld!
